mock/mock.go

```diff
@@ -209,6 +209,12 @@ func (m *Mock) On(methodName string, arguments ...interface{}) *Call {
 	return c
 }
 
+// IsCallExpected checks if a method call is expected.
+func (m *Mock) IsCallExpected(method string, arguments ...interface{}) bool {
+	found, _ := m.findExpectedCall(method, arguments...)
+	return found >= 0
+}
```

Comment:
It exposes querying the internal state of the mock, which is unnecessary for the purpose of the package. As soon as we expose it, people will be able to abuse it and we must keep it to maintain backwards compatibility.